From 1a4b231fdde7e744f9d918769d516f21e2498959 Mon Sep 17 00:00:00 2001
From: ayaka <ayaka@soulik.info>
Date: Fri, 17 Apr 2015 11:49:21 +0800
Subject: [PATCH 10/12] v4l2: remove a deadlock in videoenc framework
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It is no need to lock the encoder twice. After all the reminded
buffers have been processed, the encoder should be lock.
This patch comes from  Frédéric Sureau<frederic.sureau@vodalys.com>.

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 sys/v4l2/gstv4l2videoenc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index 74c4ff0..3e4c4d7 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -324,7 +324,6 @@ gst_v4l2_video_enc_finish (GstVideoEncoder * encoder)
         (self->v4l2output->pool), &buffer);
     gst_buffer_unref (buffer);
   }
-  GST_VIDEO_ENCODER_STREAM_LOCK (encoder);
 
   /* and ensure the processing thread has stopped in case another error
    * occured. */
-- 
2.7.4

